ReportUtilityBills/measure.rb

```diff
@@ -0,0 +1,785 @@
+    return 'Utility Bills Report'
+  end
+
+  # human readable description
+  def description
+    return 'TODO'
+  end
+
+  # human readable description of modeling approach
+  def modeler_description
+    return 'TODO'
+  end
+
+  # define the arguments that the user will input
+  def arguments(model = nil)
+    args = OpenStudio::Measure::OSArgumentVector.new
+
+    format_chs = OpenStudio::StringVector.new
+    format_chs << 'csv'
+    format_chs << 'json'
+    arg = OpenStudio::Measure::OSArgument::makeChoiceArgument('output_format', format_chs, false)
+    arg.setDisplayName('Output Format')
+    arg.setDescription('The file format of the annual (and timeseries, if requested) outputs.')
+    arg.setDefaultValue('csv')
+    args << arg
+
+    # electricity_bill_type_choices = OpenStudio::StringVector.new
+    # electricity_bill_type_choices << 'Simple'
+    # electricity_bill_type_choices << 'Detailed'
+
+    # arg = OpenStudio::Measure::OSArgument::makeChoiceArgument('electricity_bill_type', electricity_bill_type_choices, true)
+    # arg.setDisplayName('Electricity: Simple or Detailed')
+    # arg.setDescription("Choose either 'Simple' or 'Detailed'. If 'Simple' is selected, electric utility bills are calculated based on user-defined fixed charge and marginal rate. If 'Detailed' is selected, electric utility bills are calculated based on either: a tariff from the OpenEI Utility Rate Database (URDB), or a real-time pricing rate.")
+    # arg.setDefaultValue('Simple')
+    # args << arg
+
+    # utility_rate_type_choices = OpenStudio::StringVector.new
+    # utility_rate_type_choices << 'Autoselect OpenEI'
+    # utility_rate_type_choices << 'Sample Real-Time Pricing Rate'
+    # utility_rate_type_choices << 'Sample Tiered Rate'
+    # utility_rate_type_choices << 'Sample Time-of-Use Rate'
+    # utility_rate_type_choices << 'Sample Tiered Time-of-Use Rate'
+    # utility_rate_type_choices << 'User-Specified'
+
+    # arg = OpenStudio::Measure::OSArgument::makeChoiceArgument('electricity_utility_rate_type', utility_rate_type_choices, false)
+    # arg.setDisplayName('Electricity: Utility Rate Type')
+    # arg.setDescription("Type of the utility rate. Required if electricity bill type is 'Detailed'.")
+    # arg.setDefaultValue('Autoselect OpenEI')
+    # args << arg
+
+    # arg = OpenStudio::Measure::OSArgument::makeStringArgument('electricity_utility_rate_user_specified', false)
+    # arg.setDisplayName('Electricity: User-Specified Utility Rate')
+    # arg.setDescription("Absolute/relative path of the json. Relative paths are relative to the HPXML file. Required if utility rate type is 'User-Specified'.")
+    # args << arg
+
+    arg = OpenStudio::Measure::OSArgument::makeDoubleArgument('electricity_fixed_charge', true)
+    arg.setDisplayName('Electricity: Fixed Charge')
+    arg.setUnits('$/month')
+    arg.setDescription('Monthly fixed charge for electricity.')
+    arg.setDefaultValue(12.0)
+    args << arg
```

Comment:
Can we make all arguments optional instead of required? It's especially weird to require inputs for coal, propane, PV, etc. when a building might not have any of these things.

If an optional argument like `coal_marginal_rate` isn't provided but we see there is coal production in the home, I suggest we give a warning but continue to produce an output file (with zero for the coal $).

Reply:
With defaults (see below), these rates will always be provided.